python/pyspark/sql/connect/functions.py

```diff
@@ -388,7 +390,7 @@ def rand(seed: Optional[int] = None) -> Column:
     if seed is not None:
         return _invoke_function("rand", lit(seed))
     else:
-        return _invoke_function("rand")
+        return _invoke_function("rand", lit(random.randint(0, sys.maxsize)))
```

Comment:
These are spark connect functions. Do we also need to update `python/pyspark/sql/functions.py`?

Reply:
No, we don't. The sql Python functions just invoke the Scala implementations, which has been fixed.

BTW, I covered sql Python functions with a test: https://github.com/apache/spark/pull/42997/files#diff-a4984706554eae562eb0c7a66d284b1855d9bc463b03474f73493c7694244b66R1356-R1366